influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -1575,6 +1576,105 @@ mod tests {
+                        {tbl_name},warehouse=us-east,room=02a,device=30003 reading=32\n\
+                        "
+                    ),
+                    time_seconds: 2,
+                },
+                // This write will trigger the snapshot:
+                TestWrite {
+                    lp: format!(
+                        "\
+                        {tbl_name},warehouse=us-east,room=01a,device=10001 reading=35\n\
+                        {tbl_name},warehouse=us-east,room=01b,device=10002 reading=24\n\
+                        {tbl_name},warehouse=us-east,room=02a,device=30003 reading=30\n\
+                        "
+                    ),
+                    time_seconds: 3,
+                },
+            ],
+        )
+        .await;
+
+        // Wait for snapshot to be created, once this is done, then the parquet has been persisted:
+        verify_snapshot_count(1, &wbuf.persister).await;
+
+        // get the path for the created parquet file:
+        let persisted_files = wbuf.persisted_files().get_files(db_name, tbl_name);
+        assert_eq!(1, persisted_files.len());
+        let path = ObjPath::from(persisted_files[0].path.as_str());
+
+        // check the number of requests to that path before making a query:
+        // there should be one get request, made by the cache oracle:
+        assert_eq!(1, test_store.get_request_count(&path));
+        // there should be no get_range requests, since those are only made during query planning:
+        assert_eq!(0, test_store.get_range_request_count(&path));
+
+        let batches = get_table_batches(&wbuf, db_name, tbl_name, &ctx).await;
+        assert_batches_sorted_eq!(
+            [
+                "+--------+---------+------+----------------------+-----------+",
+                "| device | reading | room | time                 | warehouse |",
+                "+--------+---------+------+----------------------+-----------+",
+                "| 10001  | 35.0    | 01a  | 1970-01-01T00:00:03Z | us-east   |",
+                "| 10001  | 36.0    | 01a  | 1970-01-01T00:00:01Z | us-east   |",
+                "| 10001  | 37.0    | 01a  | 1970-01-01T00:00:02Z | us-east   |",
+                "| 10002  | 24.0    | 01b  | 1970-01-01T00:00:03Z | us-east   |",
+                "| 10002  | 28.0    | 01b  | 1970-01-01T00:00:02Z | us-east   |",
+                "| 10002  | 29.0    | 01b  | 1970-01-01T00:00:01Z | us-east   |",
+                "| 30003  | 30.0    | 02a  | 1970-01-01T00:00:03Z | us-east   |",
+                "| 30003  | 32.0    | 02a  | 1970-01-01T00:00:02Z | us-east   |",
+                "| 30003  | 33.0    | 02a  | 1970-01-01T00:00:01Z | us-east   |",
+                "+--------+---------+------+----------------------+-----------+",
+            ],
+            &batches
+        );
+
+        // counts should not change, since requests for this parquet file hit the cache:
```

Comment:
and we know without the cache these numbers go up? 😁

Reply:
I was thinking that as well, going to see if its easy to add that check to this PR.

Reply:
This should be addressed by https://github.com/influxdata/influxdb/pull/25411/commits/d97a6d98fcfc4c0a47f8977c0f950454033cb4e1 and https://github.com/influxdata/influxdb/pull/25411/commits/5c63c0af10dac4fb9132ecef0a24748b7196501a.

As I mentioned in the call earlier, requests through `iox_query`/`datafusion` seem to all be `get_range` requests, but nonetheless, the cache is set up to catch those as well, so all good...